Pin pynos version in vdx pack's requirements.txt

edonohue on #brocade:

as heads up I did fresh install of BWC and tried to install the VDX pack and there appears to be a bug in pycparser 2.15 that is preventing cryptography from building to meet pynos requirements.

pycparser 2.14 seems to work

[08:56]  
I get "AssertionError: sorry, but this version only supports 100 named groups"

[08:57]  
https://github.com/pyca/cryptography/issues/3187


lakstorm [08:59]  
Where is pynos code? github or brocade ghetto?

[09:00]  
https://github.com/BRCDcomm/pynos/blob/master/setup.py#L14

[09:01]  
Doesn’t seem like there is a direct dep on cryptography?

edonohue [09:03]  
probably ncclient -> paramiko -> cryptography

lakstorm [09:09]  
https://github.com/paramiko/paramiko/blob/master/tox-requirements.txt#L2

[09:09]  
Yep, it’s annoying when people just do >=1.1.

[09:10]  
edonohue one thing you could do is set cryptography to a specific working version pack’s requirements.txt and build the pack virtualenv by running st2 run packs.setup_virtualenv pack=vdx

mstone [09:11]  
@lakstorm we should probably do this for the vdx pack in contrib.

--- a/packs/vdx/requirements.txt
+++ b/packs/vdx/requirements.txt
@@ -1,1 +1,2 @@
-pynos
+pynos==1.3.20
+pycparser==2.14

